ee/tables/jwt/table.go

```diff
@@ -0,0 +1,195 @@
+const (
+	Valid   = "VALID"
+	Invalid = "INVALID"
+	Unknown = "UNKNOWN"
+)
+
+// Values for include_raw_jwt column.
+var (
+	allowedIncludeValues = []string{"true", "false"}
+)
+
+// Created errors here to handle switching the verified value depending on the returned error.
+var (
+	ErrMissingKeyId     = errors.New("no key id found in the JWT header")
+	ErrMatchingKeyId    = errors.New("no key id matched the JWT header key id")
+	ErrParsingPemBlock  = errors.New("error parsing PEM block containing the public key")
+	ErrParsingPublicKey = errors.New("error parsing the public key from the PEM block")
+)
+
+type Table struct {
+	slogger *slog.Logger
+}
+
+func TablePlugin(slogger *slog.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+		table.TextColumn("signing_keys"),
+		table.TextColumn("include_raw_jwt"),
+	)
+
+	t := &Table{
+		slogger: slogger.With("table", "kolide_jwt"),
+	}
+
+	return table.NewPlugin("kolide_jwt", columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+	if len(paths) < 1 {
+		return nil, fmt.Errorf("kolide_jwt requires at least one path to be specified")
+	}
+
+	for _, path := range paths {
+		for _, keyJSON := range tablehelpers.GetConstraints(queryContext, "signing_keys", tablehelpers.WithDefaults("")) {
+			for _, include_raw_jwt := range tablehelpers.GetConstraints(queryContext, "include_raw_jwt", tablehelpers.WithAllowedValues(allowedIncludeValues), tablehelpers.WithDefaults("false")) {
+				for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+					rawData, err := JWTRaw(path)
+					if err != nil {
+						t.slogger.Log(ctx, slog.LevelInfo, "error reading JWT data file", "err", err)
+						continue
+					}
+
+					// Parse provided JWT signing keys into an map for verification in the JWT parser
+					var keyMap map[string]string
+					if err := json.Unmarshal([]byte(keyJSON), &keyMap); err != nil {
+						t.slogger.Log(ctx, slog.LevelInfo, "error unmarshaling JWT signing keys", "err", err)
+					}
+
+					row := map[string]interface{}{"verified": Unknown}
+					token, err := jwt.ParseWithClaims(string(rawData), jwt.MapClaims{}, JWTKeyFunc(keyMap))
+					if err != nil {
+						t.slogger.Log(ctx, slog.LevelInfo, "error parsing token", "err", err)
+
+						if errors.Is(err, ErrParsingPemBlock) || errors.Is(err, ErrParsingPublicKey) {
+							row["verified"] = Invalid
+						}
+					} else {
+						row["verified"] = Valid
+					}
+
+					claims, ok := token.Claims.(jwt.MapClaims)
+					if !ok {
+						t.slogger.Log(ctx, slog.LevelInfo, "error parsing JWT claims")
+						continue
+					}
+
+					parsedClaims := map[string]interface{}{}
+					for k, v := range claims {
+						parsedClaims[k] = v
+					}
```

Comment:
Mmm, yes you're right I believe. This was an artifact from the dataflatten JWT code which I just didn't change, but I can see that it can be simplified.

Reply:
Eh. Some extra memory copies feel off, but not the end of the world. May not be worth that much time cleaning up